application/src/main/java/org/thingsboard/server/controller/AuthController.java

```diff
@@ -163,8 +163,12 @@ public void requestResetPasswordByEmail (
             @RequestBody JsonNode resetPasswordByEmailRequest,
             HttpServletRequest request) throws ThingsboardException {
         try {
+            mailService.testConnection();
```

Comment:
Why test connection on each reset password?

Reply:
@ashvayka I making test connection to hide exceptions when we specify correct email (present in database) and block execution of code if mail server not configured or not available, to throw more user-friendly message like 'Unable to connect to mail server' not 'Unable to send mail: 535-5.7.8 Username and Password not accepted. Learn more at 535 5.7.8 https://support.google.com/mail/?p=BadCredentials g2sm1957632ljn.35 - gsmtp'